File: src/Ui/eshop-genai-ui/src/features/order/orders/models/OrderDto.ts
export type OrderDto = {
  id: string
  userId: string
  status: string
  totalAmount: number
  shippingAddress: string
  items: OrderItemDto[]
  orderDate: string
  createdAt: string
  lastModified: string | null
}

export type OrderItemDto = {
  productId: string
  productName: string
  unitPrice: number
  quantity: number
  totalPrice: number
  imageUrl: string
}

export type CreateOrderRequest = {
  userId: string
  shippingAddress: string
}

File: src/Services/Order/src/Orders/Dtos/OrderDto.cs
using Order.Orders.Enums;

namespace Order.Orders.Dtos;

public record OrderDto(
    Guid Id,
    string UserId,
    OrderStatus Status,
    decimal TotalAmount,
    string ShippingAddress,
    List<OrderItemDto> Items,
    DateTime OrderDate,
    DateTime CreatedAt,
    DateTime? LastModified);


public record OrderItemDto(
    Guid ProductId,
    string ProductName,
    decimal UnitPrice,
    int Quantity,
    decimal TotalPrice,
    string ImageUrl);